### PR TITLE
Stream type host_forwarding

### LINKED ISSUE #589: Compose 'ports' are ignored when host forwarding is set
It seems like the `compose -> ports` section in an app spec is ignored, whenever `host_forwarding` is set. I have a use case, where I need host forwarding as well as mapping an additional port. Is ignoring additional ports a design decision or is it a bug?

Here's an example snippet from my app spec:

```yaml
[...]

host_forwarding:
  - host_name: www.project.local
    host_port: 80
    container_port: 8000

compose:
  ports:
    - "10100:10100"

[...]

```

### PR DESCRIPTION
**Pull request by thieman**
@jsingle For SSH servers and fun stuff. Still WIP, needs tests and docs and things. Fixes #589.

### CODE REVIEW

**Comment by thieman** [comment]
Example using this https://github.com/gamechanger/dusty-example-specs/pull/8/files

File: dusty/compiler/nginx/__init__.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 from ...constants import NGINX_MAX_FILE_SIZE, VM_IP_FROM_DOCKER
 
-def _nginx_proxy_string(port_spec):
-    return "proxy_pass http://{}:{};".format(VM_IP_FROM_DOCKER, port_spec['proxied_port'])
+def _nginx_proxy_string(port_spec, include_http=True):
```
Comment (jsingle) [inline dusty/compiler/nginx/__init__.py]: can we just kill this optional arg?

Comment (thieman) [inline dusty/compiler/nginx/__init__.py]: I'm using both variants of it

Comment (jsingle) [inline dusty/compiler/nginx/__init__.py]: I can't find anywhere that you set include_http

File: dusty/compiler/port_spec/__init__.py [hunk 2]
```diff
@@ -52,12 +61,15 @@ def get_port_spec_document(expanded_active_specs, docker_vm_ip):
         port_spec['docker_compose'][app_name] = []
         container_ports = set()
         for host_forwarding_spec in app_spec['host_forwarding']:
-            add_full_addresses(host_forwarding_spec, host_full_addresses)
-            add_container_ports(host_forwarding_spec, container_ports)
+            # These functions are just used for validating the set of specs all works together
+            _add_full_addresses(host_forwarding_spec, host_full_addresses)
+            _add_container_ports(host_forwarding_spec, container_ports)
+            if host_forwarding_spec['type'] == 'stream':
+                _add_stream_host_port(host_forwarding_spec, stream_host_ports)
```
Comment (jsingle) [inline dusty/compiler/port_spec/__init__.py]: `stream_host_ports.add(host_forwarding_spec['host_port'[)` should be fine after we validate for the conflict right?

Comment (thieman) [inline dusty/compiler/port_spec/__init__.py]: Yeah all that logic's bundled up in the helper function

File: dusty/systems/nginx/__init__.py [hunk 3]
```diff
@@ -20,6 +20,8 @@ def update_nginx_from_config(nginx_config):
     or tell it to reload its config to pick up what we've
     just written."""
     logging.info('Updating nginx with new Dusty config')
-    temp_path = tempfile.mkstemp()[1]
-    _write_nginx_config(nginx_config, temp_path)
-    sync_local_path_to_vm(temp_path, os.path.join(constants.NGINX_CONFIG_DIR_IN_VM, 'dustyNginx.conf'))
+    temp_dir = tempfile.mkdtemp()
+    _write_nginx_config(constants.NGINX_BASE_CONFIG, os.path.join(temp_dir, 'nginx.primary'))
+    _write_nginx_config(nginx_config['http'], os.path.join(temp_dir, 'dusty.http.conf'))
```
Comment (jsingle) [inline dusty/systems/nginx/__init__.py]: should we have constants for these?

Comment (thieman) [inline dusty/systems/nginx/__init__.py]: Yeah that'd be better

**Comment by jsingle** [comment]
Aww no new integration tests :(

**Comment by jsingle** [comment]
I think it looks good, reminder to squash commits before merging

File: dusty/compiler/nginx/__init__.py (hunk 1 above)
Comment (thieman) [inline dusty/compiler/nginx/__init__.py]: Yeah it was the default so I didn't need to include it. Anyway, took your suggestion

**Pull request merged**